pycalphad/core/calculate.py

```diff
@@ -256,39 +256,49 @@ def _compute_phase_values(components, statevar_dict, str_phase_local_conditions,
-        phase_output = broadcast_to(phase_output, points.shape[:-1])
-    if isinstance(phase_compositions, (float, int)):
-        phase_compositions = broadcast_to(phase_output, points.shape[:-1] + (len(pure_elements),))
-    phase_output = np.asarray(phase_output, dtype=np.float64)
-    if parameter_array_length <= 1:
-        phase_output.shape = points.shape[:-1]
+        if parameter_array_length == 0:
+            # No parameters specified
+            phase_output = np.zeros(dof.shape[0], order='C')
+            phase_record.prop_2d(phase_output, dof, output.encode('utf-8'))
+        else:
+            # Vectorized parameter arrays
+            phase_output = np.zeros((dof.shape[0], parameter_array_length), order='C')
+            phase_record.prop_parameters_2d(phase_output, dof, parameter_array, output.encode('utf-8'))
+
+        for el_idx in range(len(pure_elements)):
+            phase_record.mass_obj_2d(phase_compositions[:, el_idx], dof, el_idx)
+
+        max_tieline_vertices = len(pure_elements)
+        if isinstance(phase_output, (float, int)):
+            phase_output = broadcast_to(phase_output, points.shape[:-1])
+        if isinstance(phase_compositions, (float, int)):
+            phase_compositions = broadcast_to(phase_output, points.shape[:-1] + (len(pure_elements),))
+        phase_output = np.asarray(phase_output, dtype=np.float64)
+        if parameter_array_length <= 1:
+            phase_output.shape = points.shape[:-1]
+        else:
+            phase_output.shape = points.shape[:-1] + (parameter_array_length,)
+        phase_compositions = np.asarray(phase_compositions, dtype=np.float64)
+        phase_compositions.shape = points.shape[:-1] + (len(pure_elements),)
     else:
-        phase_output.shape = points.shape[:-1] + (parameter_array_length,)
-    phase_compositions = np.asarray(phase_compositions, dtype=np.float64)
-    phase_compositions.shape = points.shape[:-1] + (len(pure_elements),)
+        # We still need phase_output and phase_compositions to have the correct dimensions
+        # even if there were no points in case fake_points are added (as they will be
+        # concatenated along the points dimension). These arrays will be zero size for now.
+        # The size of the points dimension will be hardcoded to zero (np.atleast_2d will have the new dimension be size 1 even if the input array is size zero).
+        points = points.reshape(points.shape[:-2] + (0, points.shape[-1],))  # np.atleast_2d always makes a (1, N) even if size is zero, here we just make it consistent
+        if parameter_array_length <= 1:
+            phase_output = np.empty(points.shape[:-1], dtype=np.float64)
+        else:
+            phase_output = np.empty(points.shape[:-1] + (parameter_array_length,), dtype=np.float64)
+        phase_compositions = np.empty(points.shape[:-1] + (len(pure_elements),), dtype=np.float64)
+
     if fake_points:
         output_shape = points.shape[:-2] + (max_tieline_vertices,)
```

Comment:
This branch here can be accessed before `max_tieline_vertices` is created. `max_tieline_vertices` is only created if the phase is active and has feasible points. However, for a phase that is active but has no feasible points and `fake_points` is `True` in `calculate`, then `max_tieline_vertices` will not be defined before going to this branch. I think this can be easily fixed if line 278 is moved before the `if points.size > 0:` branch

Reply:
This can be tested in `test_calculate.py::test_calculate_raises_if_no_feasible_points_exists` if `fake_points` is set to True, where it will raise an `UnboundLocalError`